miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniApp.kt

```diff
@@ -189,16 +198,17 @@ abstract class MiniApp internal constructor() {
          * @return [MiniApp] instance
          */
         @JvmStatic
-        fun instance(settings: MiniAppSdkConfig = defaultConfig): MiniApp =
-            instance.apply { updateConfiguration(settings) }
+        fun instance(settings: MiniAppSdkConfig = defaultConfig, isTemporaryUpdate: Boolean = false): MiniApp =
```

Comment:
What is the `isTemporaryUpdate` property for? Is this because the Host App might want to temporarily set the SSL pin?

Couldn't you instead add the ssl pin to the `key` that is set for each API Client? It uses the [MiniApSdkConfig.key](https://github.com/munir-rakuten/android-miniapp/blob/feature/add_preview_code_api/miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppSdkConfig.kt#L28) for storing the ApiClient, so you could just append the SSL pin there. Then you still be able to distinguish between ApiClients using the same Host but one has an SSL pin and the other doesn't.

Reply:
@corycaywood actually, in demo app now the architecture is like, when we call `MiniApp.instance` it updates the all the api clients with the `SdkConfig` send as a parameter, previously the sdkConfig is always the the what is in `AppSettings`,
But now when we preview a `miniapp` through qr code from default host, we have to create the a `sdkConfig` with the `host id` and `subscriptionKey` but when we open the demo app again we can not use that `sdkConfig`, Because `appSettings` config and the miniapp we are loading from qrcode is different, that's when to make the qrcode preview sdkconfig temporary load the miniapp and the no store that clent so that when we open the demo app again it should use the config in the app settings.. does that make sense?

Reply:
If we require this property, is it possible to rename it to understand the usage & also, we need to update the change-log.

Reply:
@khairul-alam-licon @corycaywood or we can create another `instance` method like `instanceForPreviewCode` or `getInstanceForPreviewCode` and use that method without changing anything in the current instance method for that, i am confused which one will be better for this purpose..

Reply:
@munir-rakuten So basically when you pass a new Config to `instance` it is setting the new Config as the default?

I think this doesn't sound like the right behavior for the SDK, but it's too late to change it since it was already like this. Probably it should instead always use the values from `AndroidManifest` as the default and then calling `instance` with new values just changes the Config for that specific instance. But I think we can't change this now.

Instead, I think maybe just the naming for the property needs to be something that makes it more clear how it works. How about using `setConfigAsDefault: Boolean = true`?

Reply:
@corycaywood `So basically when you pass a new Config to instance it is setting the new Config as the default?` , answer is yes,...

`Instead, I think maybe just the naming for the property needs to be something that makes it more clear how it works. How about using setConfigAsDefault: Boolean = true?`
ok so true will be default.. if it's false then it will not update the default config..thanks